src/fsharp/FSharp.Core/iseq.fsi

```diff
@@ -0,0 +1,431 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.FSharp.Collections
+
+  open System
+  open System.Collections
+  open System.Collections.Generic
+  open Microsoft.FSharp.Core
+  open Microsoft.FSharp.Collections
+
+  [<CompilationRepresentation(CompilationRepresentationFlags.ModuleSuffix)>]
+  module ISeq =
+    module Core =
+        [<Struct; NoComparison; NoEquality>]
+        type NoValue = struct end
+
+        /// <summary>Values is a mutable struct. It can be embedded within the folder type
+        /// if two values are required for the calculation.</summary>
+        [<Struct; NoComparison; NoEquality>]
+        type Value<'a> =
+            new : a:'a -> Value<'a>
+            val mutable _1: 'a
+
+        /// <summary>Values is a mutable struct. It can be embedded within the folder type
+        /// if two values are required for the calculation.</summary>
+        [<Struct; NoComparison; NoEquality>]
+        type Values<'a,'b> =
+            new : a:'a * b:'b -> Values<'a,'b>
+            val mutable _1: 'a
+            val mutable _2: 'b
+
+        /// <summary>Values is a mutable struct. It can be embedded within the folder type
+        /// if three values are required for the calculation.</summary>
+        [<Struct; NoComparison; NoEquality>]
+        type Values<'a,'b,'c> =
```

Comment:
Values can be replaced by ValueTuple (as it is serves exactly the same purpose of evil mutable value type :-) but the proto-compiler doesn't know about it yet? Maybe? Or maybe I'm just stuffing something up.

All of these could be removed if the F# compiler made proper members of captured variables within object expressions. So that might be a possible thing to do before locking this public facing API out there...